system-addon/content-src/lib/snippets.js

```diff
@@ -296,15 +307,33 @@ class SnippetsProvider {
  */
 function addSnippetsSubscriber(store) {
   const snippets = new SnippetsProvider(store.dispatch);
-  const unsubscribe = store.subscribe(() => {
+
+  let unsubscribe;
+  let waitForInit;
+  let waitForUninit;
+
+  waitForInit = async () => {
     const state = store.getState();
-    if (state.Snippets.initialized) {
+    if (!snippets.initialized && state.Snippets.initialized) {
       if (state.Snippets.onboardingFinished) {
-        snippets.init({appData: state.Snippets});
+        unsubscribe();
+        await snippets.init({appData: state.Snippets});
+        unsubscribe = store.subscribe(waitForUninit);
       }
+    }
+  };
+
+  waitForUninit = () => {
```

Comment:
I'd recommend not bothering with separate unsubscribe events and simply have a conditional testing for each possible state i.e.

```js
if (snippets.initialized && !state.Snippets.intialized) {...} else if (!snippets.initialized && state.Snippets.initialized)
```

The difference between `snippets.initialized` and `state.Snippets.initialized` is additionally kind of hard to read, so some comments or perhaps renaming would help distinguish them.